### ISSUE TITLE
Add a way for server to answer on action by another actions

### ISSUE THREAD

**Issue by ai** [logux/logux#44]
It will be nice to have API like:

```js
let answer = await store.dispatch.sync(action)
answer => Action[]
```

On the server we can have something like:

```js
  async process (ctx) {
    await ctx.answer(action1)
    await ctx.answer(action2)
  }
```

We can use `meta.answer` to connect actions.

**Comment by Kirpich634** [comment]
Maybe it should be more like [API Meteor.methods(methods)](https://docs.meteor.com/api/methods.html)

**Comment by euaaaio** [comment]
> It will be nice to have API like:
> 
> ```js
> let answer = await store.dispatch.sync(action)
> answer => Action[]
> ```
> 
> On the server we can have something like:
> 
> ```js
>   async process (ctx) {
>     await ctx.answer(action1)
>     await ctx.answer(action2)
>   }
> ```
> 
> We can use `meta.answer` to connect actions.

I think, we shouldn't remove `meta`. Maybe `client.sync` will return `Promise<{ meta: ClientMeta; answer?: AnyAction[] }>`?

**Comment by ai** [comment]
@euaaaio yeap, we may need `meta` too.

Another way `client.request(action) //=> Promise<Actions[]>`

**Comment by euaaaio** [comment]
> @euaaaio yeap, we may need `meta` too.
> 
> Another way `client.request(action) //=> Promise<Actions[]>`

`client.request` looks like internal or private API against of `client.sync`. I even think that we can extend promise:
```
let { action, meta, answer } = client.sync(action, meta)
```
`action` and `meta` can be very useful in some case when we need to make undo.

**Comment by ai** [comment]
I worry that the answer is a rare API to keep inconsistency with `log.add`. But on other hand getting `meta` from `sync` is also rare. Do we have any other option to have an “illusion of choice”?

BTW, should we rename `answer` to `answers`?

**Comment by euaaaio** [comment]
> I worry that the answer is a rare API to keep inconsistency with `log.add`. But on other hand getting `meta` from `sync` is also rare. Do we have any other option to have an “illusion of choice”?

`sendBack` and `answer` cause a lot of question when you are just starting to learn Logux. I think, you are right, and these methods should be private. So we shouldn’t add them to the docs, only API.

Maybe we should call it `client.ask`? So, people will only use it for something special.

> BTW, should we rename `answer` to `answers`?

We ask and get the answer. We can call `Context#answer` several times, but the answer will be one and only after `processed` asking action. So, `answer` name is ok.
